lib/statsd.rb

```diff
@@ -1,250 +1 @@
-  #
-  # @param [String] stat stat name.
-  # @param [Numeric] gauge value.
-  # @example Report the current user count:
-  #   $statsd.gauge('user.count', User.count)
-  def gauge(stat, value)
-    send stat, value, GAUGE_TYPE
-  end
-
-  # Sends a timing (in ms) for the given stat to the statsd server. The
-  # sample_rate determines what percentage of the time this report is sent. The
-  # statsd server then uses the sample_rate to correctly track the average
-  # timing for the stat.
-  #
-  # @param stat stat name
-  # @param [Integer] ms timing in milliseconds
-  # @param [Integer] sample_rate sample rate, 1 for always
-  def timing(stat, ms, sample_rate=1); send stat, ms, TIMING_TYPE, sample_rate end
-
-  # Reports execution time of the provided block using {#timing}.
-  #
-  # @param stat (see #timing)
-  # @param sample_rate (see #timing)
-  # @yield The operation to be timed
-  # @see #timing
-  # @example Report the time (in ms) taken to activate an account
-  #   $statsd.time('account.activate') { @account.activate! }
-  def time(stat, sample_rate=1)
-    start = Time.now
-    result = yield
-    timing(stat, ((Time.now - start) * 1000).round(5), sample_rate)
-    result
-  end
-
-  # Sends a histogram measurement for the given stat to the statsd server. The
-  # sample_rate determines what percentage of the time this report is sent. The
-  # statsd server then uses the sample_rate to correctly track the average
-  # for the stat.
-  def histogram(stat, value, sample_rate=1); send stat, value, HISTOGRAM_TYPE, sample_rate end
-
-  private
-  def sampled(sample_rate)
-    yield unless sample_rate < 1 and rand > sample_rate
-  end
-
-  def send(stat, delta, type, sample_rate=1)
-    sampled(sample_rate) do
-      stat = stat.to_s.dup
-      stat.gsub!(/::/, ".".freeze)
-      stat.gsub!(RESERVED_CHARS_REGEX, "_".freeze)
-
-      msg = String.new
-      msg << @prefix
-      msg << stat
-      msg << ":".freeze
-      msg << delta.to_s
-      msg << "|".freeze
-      msg << type
-      if sample_rate < 1
-        msg << "|@".freeze
-        msg << sample_rate.to_s
-      end
-
-      shard = select_shard(stat)
-      shard.send(msg)
-    end
-  end
-
-  def select_shard(stat)
-    if @shards.size == 1
-      @shards.first
-    else
-      @shards[Zlib.crc32(stat) % @shards.size]
-    end
-  end
-
-  class Buffer
-    DEFAULT_BUFFER_CAP = 512
-
-    attr_reader :base_client
-    attr_accessor :flush_count
-
-    def initialize(client, buffer_cap = nil)
-      @base_client = client
-      @buffer = String.new
-      @buffer_cap = buffer_cap || DEFAULT_BUFFER_CAP
-      @flush_count = 0
-    end
-
-    def flush
-      return unless @buffer.bytesize > 0
-      @base_client.send(@buffer)
-      @buffer.clear
-      @flush_count += 1
-    end
-
-    def send(msg)
-      flush if @buffer.bytesize + msg.bytesize >= @buffer_cap
-      @buffer << msg
-      @buffer << "\n".freeze
-      nil
-    end
-  end
-end
+require "github/statsd"
```

Comment:
Keeping this file maintains backwards compatibility but makes it harder to load this alongside other gems that use the `statsd` namespace entry in the Ruby `LOAD_PATH` for their code.

```
Keiths-MacBook:Desktop keith$ ls lib/*/*.rb
lib/github/stats.rb	lib/others/stats.rb
Keiths-MacBook:Desktop keith$ cat lib/*/*.rb
puts "github stats"
puts "other stats"
Keiths-MacBook:Desktop keith$ ruby -Ilib/github -Ilib/others -e "puts $:; require 'stats'"
/Users/keith/Desktop/lib/github
/Users/keith/Desktop/lib/others
/usr/local/Cellar/rbenv/1.0.0/rbenv.d/exec/gem-rehash
/Library/Ruby/Site/2.0.0
/Library/Ruby/Site/2.0.0/x86_64-darwin15
/Library/Ruby/Site/2.0.0/universal-darwin15
/Library/Ruby/Site
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0/x86_64-darwin15
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0/universal-darwin15
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/x86_64-darwin15
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/universal-darwin15
github stats
```

The one you get will depend on the `LOAD_PATH` ordering, what do you think about removing this file and just having `github/statsd`?